### PR TITLE
remove cache options

### PR DESCRIPTION
**Pull request by twoertwein**
mypy is already called with `--no-incremental` which disables the cache. I simply added the corresponding flag for pytest, to remove this flag.

### CODE REVIEW

**Comment by Dr-Irv** [comment]
> mypy is already called with `--no-incremental` which disables the cache. I simply added the corresponding flag for pytest, to remove this flag.

I don't think it is called with `--no-incremental` when you run locally.  So I think this option should be kept.

**Comment by twoertwein** [comment]
`mypy --help` says:

> --no-incremental          Disable module cache (inverse: --incremental)

So I assume that is effectively the same as deleting the cache (but it it less code on our side to achieve it)

**Comment by Dr-Irv** [comment]
> `mypy --help` says:
> 
> > --no-incremental          Disable module cache (inverse: --incremental)
> 
> So I assume that is effectively the same as deleting the cache (but it it less code on our side to achieve it)

Yes, but I don't see that we are running with `--no-incremental` by default??

**Comment by Dr-Irv** [comment]
> That seems to be the case on main:
> 
> https://github.com/pandas-dev/pandas-stubs/blob/9072b1596e5364268648899a7adaf021a8d7e8c2/scripts/test/run.py#L7

Ah, good.  So you're right that we don't need the option!

File: pyproject.toml [hunk 1]
```diff
@@ -166,7 +145,7 @@ enable_error_code = "ignore-without-code" # same as in pandas
 allow_untyped_globals = false
 allow_redefinition = false
 local_partial_types = false
-implicit_reexport = false # pyright behaves the same
+implicit_reexport = true
```
Comment (twoertwein) [inline pyproject.toml:148]: Could re-enable this option again as mypy 0.971 seems to need `npt = numpy.typing` in either case

**Review by Dr-Irv — COMMENTED** [review]
Given that you had to change the `npt` reference because of `mypy` 0.971, can you bump the version in `pyproject.toml` ?

File: scripts/test/run.py [hunk 2]
```diff
@@ -14,7 +13,7 @@ def pyright_src():
 
 
 def pytest():
-    cmd = ["pytest"]
+    cmd = ["pytest", "--cache-clear"]
```
Comment (Dr-Irv) [inline scripts/test/run.py:16]: double checking that not clearing the cache will always work.

File: scripts/test/run.py [hunk 3]
```diff
@@ -61,13 +60,3 @@ def restore_src():
...
-
-
-def clean_mypy_cache():
-    if Path(".mypy_cache").exists():
-        shutil.rmtree(".mypy_cache")
-
-
-def clean_pytest_cache():
-    if Path(".mypy_cache").exists():
-        shutil.rmtree(".pytest_cache")
```
Comment (Dr-Irv) [inline scripts/test/run.py:73]: the test was wrong here, so the `-c` never worked!

Comment (twoertwein) [reply scripts/test/run.py:73]: hahaha :) I didn't notice that when I removed it :)

File: scripts/test/run.py (hunk 2 above)
Comment (twoertwein) [reply scripts/test/run.py:16]: Locally, I do not need to clear the cache. Happy to remove it. The CI will anyways have a clean plate every time it runs.

Comment (Dr-Irv) [reply scripts/test/run.py:16]: So I see.  Now your proposal is to *always* clear the cache.  I think that's a good idea.  Equivalent to what will happen with `mypy` where we *always* use `--no-incremental`

**Review by Dr-Irv — APPROVED** [review]
Thanks @twoertwein

**Pull request merged**